Added sample for TF Flask tutorial

betsy-lichtenberg opened the pull request:
Intending for this sample to live at https://cloud.devsite.corp.google.com/docs/terraform/terraform-in-google-cloud-quickstart, which will be replacing this community post: https://cloud.google.com/community/tutorials/getting-started-on-gcp-with-terraform

```release-note:none
```

modular-magician commented:
Hello!  I am a robot who works on Magic Modules PRs.

I have detected that you are a community contributor, so your PR will be assigned to someone with a commit-bit on this repo for initial review.

Thanks for your contribution!  A human will be with you soon.

@slevenick, please review this PR or find an appropriate assignee.

slevenick reviewed (changes requested):
It looks like this doesn't generate correctly. Not entirely sure why, but there are compilation errors due to this change

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 211 insertions(+), 128 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 211 insertions(+), 128 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-5839-old..auto-pr-5839) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 57 insertions(+))

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 226 insertions(+), 128 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 226 insertions(+), 128 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-5839-old..auto-pr-5839) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 74 insertions(+))

slevenick commented:
Still failing with 
```
google-beta/resource_cgc_snippet_generated_test.go:25:29: expected '(', found '-'
?   	github.com/hashicorp/terraform-provider-google-beta	[no test files]
?   	github.com/hashicorp/terraform-provider-google-beta/version	[no test files]
```

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 228 insertions(+), 128 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 228 insertions(+), 128 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-5839-old..auto-pr-5839) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 76 insertions(+))

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 227 insertions(+), 128 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 227 insertions(+), 128 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-5839-old..auto-pr-5839) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 75 insertions(+))

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 228 insertions(+), 128 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 228 insertions(+), 128 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-5839-old..auto-pr-5839) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 76 insertions(+))

slevenick commented:
/gcbrun

Not sure why this isn't generating correctly

betsy-lichtenberg commented:
> 

I thought I already removed public access protection in this commit:
https://github.com/GoogleCloudPlatform/magic-modules/pull/5839/commits/6c5c1c3705ff5fec38d47d56f9e50550a000f738

Removed again.

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 227 insertions(+), 128 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 227 insertions(+), 128 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-5839-old..auto-pr-5839) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 75 insertions(+))

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 226 insertions(+), 128 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 226 insertions(+), 128 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-5839-old..auto-pr-5839) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 74 insertions(+))

c2thorn commented on mmv1/products/cgc/terraform.yaml:
```diff
@@ -16,6 +16,16 @@ overrides: !ruby/object:Overrides::ResourceOverrides
     examples:
+    # Compute
+      - !ruby/object:Provider::Terraform::Examples
+        name: "flask-google-cloud-quickstart"
```
@betsy-lichtenberg the filename and reference to it here needs to be snake case. We use this name as-is in our code, and the hyphens cause build errors

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 96 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 96 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-5839-old..auto-pr-5839) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 74 insertions(+))

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccDataprocMetastoreServiceDatasource_basic|TestAccDatasourceGoogleServiceNetworkingPeeredDnsDomain_basic|TestAccCGCSnippet_flaskGoogleCloudQuickstartExample|TestAccContainerCluster_withAuthenticatorGroupsConfig You can view the result here: https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=273401

modular-magician commented:
Tests failed during RECORDING mode: TestAccCGCSnippet_flaskGoogleCloudQuickstartExample|TestAccContainerCluster_withAuthenticatorGroupsConfig Please fix these to complete your PR

slevenick commented:
Looks like the test fails due to:

```
provider_test.go:309: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
(map[string]string) (len=2) {
(string) (len=10) "metadata.%": (string) (len=1) "1",
(string) (len=23) "metadata.startup-script": (string) (len=97) "sudo apt-get update; sudo apt-get install -yq build-essential python-pip rsync; pip install flask"
}
(map[string]string) (len=1) {
(string) (len=23) "metadata_startup_script": (string) (len=97) "sudo apt-get update; sudo apt-get install -yq build-essential python-pip rsync; pip install flask"
}
```

We will need to skip comparing this value in import by using `ignore_read_extra` in the example configuration. Likely on both metadata & metadata_startup_script

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 97 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 97 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-5839-old..auto-pr-5839) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 74 insertions(+))

slevenick commented:
/gcbrun

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 97 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 97 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-5839-old..auto-pr-5839) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF Cloud Doc Samples: [Diff](https://github.com/modular-magician/terraform-docs-samples/compare/auto-pr-5839-old..auto-pr-5839) ( 1 file changed, 74 insertions(+))

modular-magician commented:
I have triggered VCR tests in RECORDING mode for the following tests that failed during VCR: TestAccDatasourceGoogleServiceNetworkingPeeredDnsDomain_basic|TestAccCGCSnippet_flaskGoogleCloudQuickstartExample|TestAccContainerCluster_withAuthenticatorGroupsConfig|TestAccFirebaserulesRelease_BasicRelease You can view the result here: https://ci-oss.hashicorp.engineering/viewQueued.html?itemId=274914
